libraries/eosiolib/core/eosio/crypto_bls_ext.hpp

```diff
@@ -490,16 +491,33 @@ namespace detail {
         bls_g1 g1_points[2] = {0};
         bls_g2 g2_points[2] = {0};
 
-        memcpy(g1_points[0].data(), G1_ONE_NEG.data(), G1_ONE_NEG.size());
-        memcpy(g2_points[0].data(), signature_proof.data(), signature_proof.size());
+        std::memcpy(g1_points[0].data(), G1_ONE_NEG.data(), G1_ONE_NEG.size());
+        std::memcpy(g2_points[0].data(), signature_proof.data(), signature_proof.size());
 
-        memcpy(g1_points[1].data(), pubkey.data(), pubkey.size());
+        std::memcpy(g1_points[1].data(), pubkey.data(), pubkey.size());
         g2_fromMessage(pubkey, POP_CIPHERSUITE_ID, g2_points[1]);
 
         bls_gt r;
         bls_pairing(g1_points, g2_points, 2, r);
 
-        return 0 == std::memcmp(r.data(), GT_ONE.data(), sizeof(bls_gt));
+        return 0 == std::memcmp(r.data(), GT_ONE.data(), GT_ONE.size());
+    }
+
+    // pubkey and signature are assumed to be in RAW affine little-endian bytes
+    bool bls_signature_verify(const bls_g1& pubkey, const bls_g2& signature_proof, const std::string& msg) {
+        bls_g1 g1_points[2];
+        bls_g2 g2_points[2];
+
+        std::memcpy(g1_points[0].data(), detail::G1_ONE_NEG.data(), detail::G1_ONE_NEG.size());
+        std::memcpy(g2_points[0].data(), signature_proof.data(), signature_proof.size());
+
+        std::memcpy(g1_points[1].data(), pubkey.data(), pubkey.size());
+        detail::g2_fromMessage(msg, detail::CIPHERSUITE_ID, g2_points[1]);
```

Comment:
`detail::` not needed.

Reply:
It is needed. This method is outside of `detail` namespace.